spec/acceptance/iis_site_spec.rb

```diff
@@ -545,5 +545,34 @@
         remove_all_sites
       end
     end
+
+    context 'with conflicting sites' do
+      before (:all) do
+        create_path('C:\inetpub\basic')
+        @site_name = "#{SecureRandom.hex(10)}"
+        @second_site_name = "#{SecureRandom.hex(10)}"
+        create_site(@site_name, true)
+
+        @manifest = <<-HERE
+          iis_site { "#{@second_site_name}":
+            ensure          => 'started',
+            physicalpath    => 'C:\\inetpub\\basic',
+            applicationpool => 'DefaultAppPool',
+            bindings        => [
+              {
+                'bindinginformation' => "*:8080:#{@second_site_name}",
+                'protocol'           => 'http',
+              }
+            ],
+          }
+        HERE
+      end
+
+      it_behaves_like 'an idempotent resource'
```

Comment:
So I looked at the other tests and saw some similar ones already that specified `bindinginformation` like this that didn't fail.

If you go back to the top of the test file, you'll see that entire suite starts off with a `remove_all_sites` to remove the Default binding:

https://github.com/RandomNoun7/puppetlabs-iis/blob/57fd6505f2a9d70a68fcfc97abd846329439cc99/spec/acceptance/iis_site_spec.rb#L6

So you have a couple of options:

* Put this test *before* that happens to prove this works. You would also want to add an assertion to this test that confirming another site exists on 80.
* Modify the setup of this test to add a different site on 80, and assert that its there (it's state of running / stopped shouldn't matter).  Then you can verify no errors get thrown trying to create a new site because you're creating a new site bound to a different port instead of default binding it to 80 first.

Reply:
To follow up on this, @RandomNoun7 noted I missed the bit about `create_site` that actually sets up a site on 80 as part of the `before(:all)`.  So it looks like there *is* a site running on 80 prior to running the manifest, which would have previously conflicted.

For the sake of understanding the test, it would be helpful to add a couple of additional assertions around the parts that are most important for this test. To the `before` we want to add:

```
@result = resource('iis_site', @site_name)
```

So that before the `it_behaves_like 'an idempotent resource'` we can add:

```ruby
puppet_resource_should_show('port', 80)
```

Then after the idempotent check that runs the manifest:


```ruby
@result = resource('iis_site', @second_site_name)
puppet_resource_should_show('port', 8080)
```

I think it's preferable to be able to check the state outside of what `puppet resource` tells us, but if that's the easy way, it's already an established pattern in the test file.

Reply:
New tests added.